queries.sql

```diff
@@ -8,3 +8,86 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg>10.5
 SELECT name FROM animals WHERE neutered=true
 SELECT * FROM animals WHERE name <> 'Gabumon'
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.4 AND 17.3;
+
+BEGIN TRANSACTION;
+UPDATE animals
+SET species = 'unspecified';
+SELECT * FROM animals WHERE species = 'unspecified';
+ROLLBACK;
+SELECT * FROM animals WHERE species <> 'unspecified';
+
+
+BEGIN TRANSACTION;
+UPDATE animals
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+UPDATE animals
+SET species = 'pokemon'
+WHERE species IS NULL;
+COMMIT;
+
+
+BEGIN TRANSACTION;
+DELETE FROM animals;
+SELECT COUNT(*) FROM animals;
+ROLLBACK;
+SELECT COUNT(*) FROM animals;
+
+
+BEGIN TRANSACTION;
+
+-- Delete all animals born after Jan 1st, 2022
+DELETE FROM animals
+WHERE date_of_birth > '2022-01-01';
+
+-- Create a savepoint for the transaction
+SAVEPOINT my_savepoint;
+
+-- Update all animals' weight to be their weight multiplied by -1
+UPDATE animals
+SET weight_kg = -1 * weight_kg;
+
+-- Rollback to the savepoint
+ROLLBACK TO my_savepoint;
+
+-- Update all animals' weights that are negative to be their weight multiplied by -1
+UPDATE animals
+SET weight_kg = -1 * weight_kg
+WHERE weight_kg < 0;
+
+-- Commit the transaction
+COMMIT;
+
+SELECT COUNT(*) FROM animals;
+
+SELECT COUNT(*) FROM animals WHERE escape_attempts = 0;
+
+SELECT AVG(weight_kg) as avg_weight FROM animals;
+
+SELECT neutered, AVG(escape_attempts) as avg_escape_attempts
+FROM animals
+GROUP BY neutered;
+
+SELECT species, MIN(weight_kg) as min_weight, MAX(weight_kg) as max_weight
+FROM animals
+GROUP BY species;
+
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
+FROM animals
+WHERE date_of_birth BETWEEN DATE '1990-01-01' AND DATE '2000-12-31'
+GROUP BY species;
+
+-- correct wightt of Angemon 
```

Comment:
Great job @Kashif-Saleem-Ghuman 🤝

- [ ] [Optional] As stated by the previous reviewer, kindly add update the weight_kg value of `Squirtle` and add these changes to the data.sql file.